### PR TITLE
drop down item with valueString answer options selection crash fixed

### LINKED ISSUE #1542: Re-selecting an option in a drop down item with valueString answer options is causing app to crash
**Describe the bug**
An item with the type 'choice', using the drop down extension, and answer options comprised of valueStrings crashes the app when an option is re-selected. The first time the option is selected, nothing happens but when you attempt to select an other option from the same drop down, the app crashes.

**Component**
Demo app

**To Reproduce**
1. Update the file 'new-patient-registration-paginated.json' to use an item of the type 'choice' for the country attribute, remove the initial expression attribute (because of [this](https://github.com/google/android-fhir/issues/1541) issue), and add valueString answer options.
2. Run the demo app
3. Tap on a patient
4. Tap on the edit button
5. Select a country
6. Select a different country
7. The app will crash with the error: Caused by: java.lang.IllegalArgumentException: Mismatching question type CHOICE and answer type string for PR-address-country

**Expected behavior**
I expect the second country to be selected

**Screenshots**
N/A

**Smartphone (please complete the following information):**
- Device: Samsung A12
- Android version: 11
- Build number: RP1A.200720.012.A127FXXU4AUK1
- Version: 30

**Additional context**
Support for the string type for a choice widget was added in this [PR](https://github.com/google/android-fhir/pull/1399). The updated item would look like this:
```
{
              "extension": [
                {
                  "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
                  "valueCodeableConcept": {
                    "coding": [
                      {
                        "system": "http://hl7.org/fhir/questionnaire-item-control",
                        "code": "drop-down",
                        "display": "Drop down"
                      }
                    ],
                    "text": "Drop down"
                  }
                }
              ],
              "linkId": "PR-address-country",
              "definition": "http://hl7.org/fhir/StructureDefinition/Patient#Patient.address.country",
              "type": "choice",
              "answerOption": [
                {
                  "valueString": "Kenya"
                },
                {
                  "valueString": "Malawi"
                },
                {
                  "valueString": "Zambia"
                }
              ],
              "required": true,
              "text": "Country",
              "_text": {
                "extension": [
                  {
                    "extension": [
                      {
                        "url": "lang",
                        "valueCode": "sw"
                      },
                      {
                        "url": "content",
                        "valueString": "Nchi"
                      }
                    ],
                    "url": "http://hl7.org/fhir/StructureDefinition/translation"
                  }
                ]
              }
            }
          ]
        }
```

**Would you like to work on the issue?**
Maybe the author of the [PR](https://github.com/google/android-fhir/pull/1399) can look into resolving this

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1542 

**Description**
1. App crash on drop down item re-selection with valueString as Answer option
2. 2. test cases added for same

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt [hunk 1]
```diff
@@ -57,7 +57,7 @@ internal object QuestionnaireItemDropDownViewHolderFactory :
         val adapter =
           ArrayAdapter(context, R.layout.questionnaire_item_drop_down_list, answerOptionString)
         autoCompleteTextView.setText(
-          questionnaireItemViewItem.answers.singleOrNull()?.valueCoding?.display ?: ""
+          questionnaireItemViewItem.answers.singleOrNull()?.value?.toString() ?: ""
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt:60]: the logic for display value coding object needs to be present here: if there's a display field, we should show that, and if there isn't, we fall back to the coding string itself.

this needs to be consistent for the entire library. but i feel currently we're not doing a good job making sure that the same logic is called everywhere.

we probably should have some kind of extension function on `Type` to unify this logic for different types, so you can call `value?.displayValue` here.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/validation/QuestionnaireResponseValidator.kt [hunk 2]
```diff
@@ -276,7 +276,7 @@ object QuestionnaireResponseValidator {
           "Mismatching question type $questionnaireItemType and answer type $answerType for $linkId"
         }
       Questionnaire.QuestionnaireItemType.CHOICE ->
-        require(answerType == "Coding") {
+        require(answerType == "Coding" || answerType == "string") {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/validation/QuestionnaireResponseValidator.kt:279]: matching answer type with the widget type choice/open should probably be in a function (see line 283 below which is the same).

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/validation/QuestionnaireResponseValidator.kt (hunk 2 above)
Comment (PallaviGanorkar) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/validation/QuestionnaireResponseValidator.kt]: Done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt (hunk 1 above)
Comment (PallaviGanorkar) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDropDownViewHolderFactory.kt]: Done

**Pull request merged**